Documentation/design/operatorgroups.md

```diff
@@ -0,0 +1,221 @@
+# Operator Multitenancy with OperatorGroups
+
+An `OperatorGroup` is an OLM resource that provides rudimentary multitenant configuration to OLM installed operators.
+
+## OperatorGroup Overview
+
+* An `OperatorGroup` selects a set of target namespaces in which to generate required RBAC access for its member operators.
+* The set of target namespaces is provided via a comma-delimited string stored in the `olm.targetNamespaces` annotation. This annotation is applied to member operator's deployments and is accessible through [The Downward API](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/#the-downward-api)
+* An operator is said to be a [member of an `OperatorGroup`](#operatorgroup-membership) if its CSV exists in the same namespace as the `OperatorGroup` and its CSV's [`InstallModes` support the set of namespaces targetted by the `OperatorGroup`](#installmodes-and-supported-operatorgroups)
+* In order to transition, a CSV must be an active member of an `OperatorGroup` that has no [provided API conflicts with intersecting `OperatorGroups`](#operatorgroup-intersection)
+
+## OperatorGroup Membership
+
+An operator defined by CSV `csv-a` is said to be a _member_ of `OperatorGroup` `op-a` in namespace `ns-a` if both of the following hold:
+* `op-a` is the only `OperatorGroup` in `ns-a`
+* `csv-a`'s `InstallMode`s support `op-a`'s target namespace set 
+
+### TooManyOperatorGroups
+
+If there exists more than one `OperatorGroup` in a single namespace, any CSV created in that namespace will transition to a failure state with reason `TooManyOperatorGroups`. CSVs in a failed state for this reason will transition to pending once the number of `OperatorGroup`s in their namespaces reaches one.
+
+### InstallModes and Supported OperatorGroups
+
+An `InstallMode` consists of an `InstallModeType` field and a boolean `Supported` field. A CSV's spec can contain a set of `InstallModes` of four distinct `InstallModeTypes`: 
+* `OwnNamespace`: If supported, the set of namespaces targetted by an `OperatorGroup` must contain the namespace the Operator is to be installed in.
+* `SingleNamespace`: If supported, the set of namespaces targetted by an `OperatorGroup` can be of length 1.
+* `MultiNamespace`: If supported, the set of namespaces targetted by an `OperatorGroup` can be of length >= 1. Any Operator supporting `MultiNamespace` implicitly supports `SingleNamespace` as well.
+* `AllNamespaces`: If supported, the Operator can support an `OperatorGroup` that selects all namespaces with a targetted set = [""]. Any Operator supporting `AllNamespaces` implicity supports `OwnNamespace` as well. 
```

Comment:
I don’t think we would install an operator with ‘AllNamespaces’ into a group watching one namespaces, so I think that this extra comment makes sense

Reply:
This just means that any CSV supporting `AllNamespaces` must also support `OwnNamespace` - For reference OwnNamespace doesn't imply SingleNamespace.

Reply:
But can't I write an operator that starts up and watches all namespaces, but can't be deployed in a configuration that watches only its own namespace?

Reply:
In the current implementation, `OwnNamespace` is an additional restriction on an OperatorGroup’s namespace set. More specifically, it means that a CSV can be a member of an OperatorGroup whose target namespace set includes the CSV’s namespace. This rule applies to `Single`, `Multi`, and `All`, rather than just applying to `Single` (see https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/api/apis/operators/v1alpha1/clusterserviceversion.go#L99).

We can change this logic to match your definition (It makes sense to) - but I think a change like that warrants a follow up PR.